src/vm/eventpipe.cpp

```diff
@@ -360,6 +378,12 @@ void EventPipe::Disable()
         // Disable the profiler.
         SampleProfiler::Disable();
 
+        // Log the process information event.
+        s_pEventSource->SendProcessInfo(s_pCommandLine);
```

Comment:
Why wait with this until `Disable`? imho the sooner we have this information, the better (I would move it to Enable).

I am not sure who is going to call [ProcessName](https://github.com/Microsoft/perfview/blob/121cb41fdca1382dfb7fe5d958d9c4a5abe6c739/src/TraceEvent/EventPipe/EventPipeEventSource.cs#L121) and when (after processing all events or maybe in the meantime?)

Reply:
I am waiting until Disable in case the circular buffer wraps.  This guarantees that this event doesn't get discarded.

With respect to TraceEvent - TraceLog already does a pass over the data before you can use it, which means that users won't notice that this isn't part of the header.  That's actually the point of TraceLog - to allow you to fill data structures with information that is only available after you need it.  Thus, TraceLog does a single pass and then the user of the TraceLog does a second pass.